<!DOCTYPE html>
<html lang="en">

<head>
	<meta charset="utf-8" />
	<link rel="icon" href="../favicon.png" />
	<meta name="viewport" content="width=device-width" />
	
		<link href="../_app/immutable/assets/0.CUHd3XmB.css" rel="stylesheet">
		<link href="../_app/immutable/assets/4.CqroqSfD.css" rel="stylesheet">
		<link rel="modulepreload" href="../_app/immutable/entry/start.CugXPWPO.js">
		<link rel="modulepreload" href="../_app/immutable/chunks/DQoubQey.js">
		<link rel="modulepreload" href="../_app/immutable/chunks/CzYMjeXN.js">
		<link rel="modulepreload" href="../_app/immutable/chunks/BX8MYmal.js">
		<link rel="modulepreload" href="../_app/immutable/entry/app.CqO8dlCL.js">
		<link rel="modulepreload" href="../_app/immutable/chunks/PPVm8Dsz.js">
		<link rel="modulepreload" href="../_app/immutable/chunks/CxZAt6QE.js">
		<link rel="modulepreload" href="../_app/immutable/chunks/Bx5urKML.js">
		<link rel="modulepreload" href="../_app/immutable/chunks/Pn26e8Hi.js">
		<link rel="modulepreload" href="../_app/immutable/chunks/D1FBDeXU.js">
		<link rel="modulepreload" href="../_app/immutable/chunks/DW64Ykzn.js">
		<link rel="modulepreload" href="../_app/immutable/nodes/0.CMdwUE9Y.js">
		<link rel="modulepreload" href="../_app/immutable/chunks/BRtrnFv7.js">
		<link rel="modulepreload" href="../_app/immutable/chunks/DbovNYc0.js">
		<link rel="modulepreload" href="../_app/immutable/nodes/4.C8o-XHJl.js">
</head>

<body data-sveltekit-preload-data="hover">
	<div style="display: contents;"><!--[--><!--[--><!----><div class="wrapper svelte-12qhfyh"><header class="svelte-8t269v"><nav class="svelte-8t269v"><a href="/"><h1>Chris' Blog</h1></a> <a href="/about">About me</a></nav></header><!----> <main class="main-blob svelte-12qhfyh"><!----><article class="svelte-14u6r3i"><h1 style="text-align:center; font-size:24px">A little foray into the virtual cell challenge.</h1> <!----><p>The virtual cell challenge is a brand new challenge released by the <a href="https://arcinstitute.org/" rel="nofollow">Arc Institute</a> to help progress single cell predictions against perturbations. This research is crucial into our understanding of how cells behave under perturbations and allows us to create novel cells. It comes alongside their own release of <a href="https://github.com/ArcInstitute/state" rel="nofollow">State</a>, a model to do just this.</p> <blockquote><p>Understanding, predicting, and ultimately programming how cells respond to internal cues and external stimuli is a fundamental challenge in biology.</p></blockquote> <p>From
The data is split into 3. Training with 150 perturbations, Validation set with 50 perturbations and the final test set with 50 perturbations.</p> <h3>Timeline</h3> <p>The absence to the</p> <h3>The PDS spiral</h3> <p>Perturbation Discrimination Score is one of the core evaluation metrics and as it turns out, became almost the sole decider of the whole competition. This was due to a few factors.</p> <h4>Scoring is skewed towards PDS</h4> <p>The 3 evaluation metrics are Mean Absolute Error, Differential Expression Score (DES) and Perturbation Discrimination Score (PDS)</p> <p>Metrics are measured based on baseline value. The baselines are created using the aggregate cell-mean change.
The baseline ended up being 0.027 for MAE, 0.106 for DES and 0.514 for PDS.</p> <img src="/_app/immutable/assets/vcc_pds_dominates_scoring.CJqLgBXl.png" width="" alt="Graphs by Fleetwood in discord https://discord.com/channels/1385383995704999947/1389397364401373335/1424753215131357286"/> Because DES and PDS are then scaled so that 1 is perfect while MAE is scaled so that 0 is perfect. This meant the gradient of the points value is vastly different. A single point increase in PDS amounts to much more than either DES or MAE. The added fact that any score under the baseline is cropped to zero, it meant MAE is almost useless (regardless of what the organisers are trying to convince us). <blockquote><p>We made a design choice this year not to penalize models that perform worse than the baseline on any one metric because they may still outperform on other metrics. We understand there is a downside to this approach: as the current leaderboard shows, some top-performing models are not doing well at all on MAE. But that does not mean MAE can’t be a differentiating factor, if some Challengers find a way to do better on that metric. <a href="https://discord.com/channels/1385383995704999947/1423038641764175882/1423746443583291516" rel="nofollow">https://discord.com/channels/1385383995704999947/1423038641764175882/1423746443583291516</a></p></blockquote> <p>Note PDS also already caused headache in the middle-end stage of the competition where a small code inaccuracy of an misused <code>abs</code> operator caused the metric to consider only absolute sizes of differences and not the direction. This was correctly changed by the organiser. <a href="https://discord.com/channels/1385383995704999947/1387880200880132359/1423037528046440652" rel="nofollow">https://discord.com/channels/1385383995704999947/1387880200880132359/1423037528046440652</a></p> <h4>Pseudo-bulking</h4> <p>PDS is calculated on the pseudo-bulk level. This means that all submitted cells of the same cell lines are grouped and average before it goes into the PDS calculation. This in turn makes the single-cell nature of the expression profiles a distraction for this metric. DES does look at single cell, but</p> <h3>Gaming the leaderboard</h3> <p>The validation and test set’s non-targeting cells are not released.</p> <h3>My small attempt</h3> <p>My attempt is to leverage a simple encoder-only transformer based on the following:</p> <p><code>Measured Gene ESM embedding</code> + <code>Perturbed Gene ESM embedding</code> + <code>Original control value random</code> => <code>Predicted value</code></p> <p>I never managed to get a proper validation workflow working with the validation set (which is a persistent problem amongst other participants as the control cells are different in the test sets). Hence as I was running out of time, I ended up only training via the loss function.</p> <img src="/_app/immutable/assets/wandb_train_loss.CNmHbZsI.png" width="600" alt="Wandb Train loss"/> <img src="/_app/immutable/assets/wandb_val_loss.DfhkaVG3.png" width="600" alt="Wandb Val loss"/> <p>On inference, I also pick random control cells from the original set. The charts look nice but the performance was not!</p> <h3>Aftermath</h3><!----></article><!----><!----></main> <footer class="svelte-178dl3j"><nav class="svelte-178dl3j"><a href="https://github.com/lcmchris"><img class="icon svelte-178dl3j" src="/github-mark.svg" alt="GitHub"/></a> <a href="mailto:lcmchristopher@gmail.com"><img class="icon svelte-178dl3j" src="/email.svg" alt="Email"/></a> <a href="/rss.xml">RSS</a> <a href="https://follow.it/chris-l-s-blog-about-things?leanpub">Subscribe via Email</a></nav></footer><!----></div><!----><!--]--> <!--[!--><!--]--><!--]-->
			
			<script>
				{
					__sveltekit_ax00jb = {
						base: new URL("..", location).pathname.slice(0, -1)
					};

					const element = document.currentScript.parentElement;

					Promise.all([
						import("../_app/immutable/entry/start.CugXPWPO.js"),
						import("../_app/immutable/entry/app.CqO8dlCL.js")
					]).then(([kit, app]) => {
						kit.start(app, element, {
							node_ids: [0, 4],
							data: [null,null],
							form: null,
							error: null
						});
					});
				}
			</script>
		</div>
</body>

</html>